sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/unit/tests/inputs/ConfigurableEmployeeInputFormat.java

```diff
@@ -0,0 +1,189 @@
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.beam.sdk.io.hadoop.inputformat.unit.tests.inputs;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.beam.sdk.values.KV;
+import org.apache.hadoop.conf.Configurable;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+/**
+ * <p>
+ * This is a InputFormat for reading employee data which is available in the form of
+ * {@code List<KV>} as {@linkplain ConfigurableEmployeeRecordReader#employeeDataList
+ * employeeDataList}. {@linkplain ConfigurableEmployeeRecordReader#employeeDataList
+ * employeeDataList} is populated using {@linkplain TestEmployeeDataSet#populateEmployeeData()}.
+ * <p>
+ * This is input to test reading using HadoopInputFormatIO if InputFormat implements Configurable.
+ * Known InputFormats which implements Configurable are DBInputFormat, TableInputFormat etc.
+ */
+public class ConfigurableEmployeeInputFormat extends InputFormat<Text, Employee>
+    implements Configurable {
+  private long NUMBER_OF_SPLITS;
+  private long NUMBER_OF_RECORDS_IN_EACH_SPLIT;
+  public boolean isConfSet = false;
+
+  public ConfigurableEmployeeInputFormat() {}
+
+  @Override
+  public Configuration getConf() {
+    return null;
+  }
+
+  /**
+   * Set configuration properties such as number of splits and number of records in each split.
+   */
+  @Override
+  public void setConf(Configuration conf) {
+    NUMBER_OF_SPLITS = conf.getLong("split.count", 3L);
+    NUMBER_OF_RECORDS_IN_EACH_SPLIT = conf.getLong("split.records.count", 5L);
+    isConfSet = true;
+  }
+
+  @Override
+  public RecordReader<Text, Employee> createRecordReader(InputSplit split,
+      TaskAttemptContext context) throws IOException, InterruptedException {
+    return new ConfigurableEmployeeRecordReader();
+  }
+
+  /**
+   * Returns InputSPlit list of {@link ConfigurableEmployeeInputSplit}. 
+   * Throws exception if {@link #setConf()} is not called.
+   */
+  @Override
+  public List<InputSplit> getSplits(JobContext context) throws IOException, InterruptedException {
+    if (!isConfSet) {
+      throw new IOException("Configuration is not set.");
```

Comment:
so there's a lot of boiler plate code in this InputFormat. A lot of it doesn't really seem necessary (ie, why do we need to implement a real reader inside of this code?)

is it important to actually test that the HIF does anything besides call setConf before calling getSplits? if so, then you could throw a different exception inside of getSplits when setConf *has* been called, and just terminate the test there. The test would just start the HIF reading from the inputformat, and expect to get back the good exception, not the bad exception. 

Does that make sense?